app/queues/dns/add-record-flow.server.ts

```diff
@@ -0,0 +1,105 @@
+import { FlowProducer } from 'bullmq';
+import { redis } from '~/lib/redis.server';
+import { buildDomain } from '~/utils';
+import { dnsUpdateQueueName } from './workers/dns-update-worker.server';
+import { checkDnsStatusQueueName } from './workers/poll-dns-status-worker.server';
+import { syncDbStatusQueueName } from './workers/sync-db-status-worker.server';
+import { createUserRecord } from '~/models/record.server';
+
+import type { FlowJob } from 'bullmq';
+import type { Record } from '@prisma/client';
+import type { DnsUpdaterData } from './workers/dns-update-worker.server';
+import type { DbRecordSynchronizerData } from './workers/sync-db-status-worker.server';
+
+export type Name = { subdomain: string } | { fqdn: string };
+interface BaseAddDnsRequestData {
+  username: Record['username'];
+  type: Record['type'];
+  value: Record['value'];
+  description?: Record['description'];
+  course?: Record['course'];
+  ports?: Record['ports'];
+}
+export type AddDnsRequestData = BaseAddDnsRequestData & Name;
+
+export enum WorkType {
+  create = 'create',
+  update = 'update',
+  delete = 'delete',
+}
+
+const flowProducer = new FlowProducer({ connection: redis });
+
+export const addDnsRequest = async (data: AddDnsRequestData) => {
+  const { username, type, value } = data;
+
+  let subdomain: string;
+  if ('subdomain' in data) {
+    subdomain = data.subdomain;
+  } else {
+    throw new Error('fqdn is not expected');
```

Comment:
You could use [`tiny-invariant`](https://www.npmjs.com/package/tiny-invariant) (already in our deps) here:

```ts
invariant('subdomain' in data, 'expected subdomain vs. fqdn');
const subdomain = data.subdomain;
```